File: app/src/main/java/com/reelme/app/view/RelegionActivity.java
package com.reelme.app.view;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.reelme.app.R;
import com.reelme.app.adapter.RelegiousRecyclerViewAdapter;
import com.reelme.app.databinding.RelegiousLayoutBinding;
import com.reelme.app.model.Flight;

import java.util.ArrayList;
import java.util.List;

public class RelegionActivity extends AppCompatActivity {
    private RelegiousRecyclerViewAdapter adapter;
    private RelegiousLayoutBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = DataBindingUtil.setContentView(this, R.layout.relegious_layout);
        binding.flightsRv.setLayoutManager(new LinearLayoutManager(this));
        binding.flightsRv.addItemDecoration(
                new DividerItemDecoration(this, DividerItemDecoration.VERTICAL));

        RelegiousRecyclerViewAdapter adapter =
                new RelegiousRecyclerViewAdapter(prepareData(), this);
        binding.flightsRv.setAdapter(adapter);
        binding.nextBtn.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                startActivity(new Intent(RelegionActivity.this, FragmentHobbies.class));
            }
        });



    }

    public List<Flight> prepareData(){
        List<Flight> flights = new ArrayList<>();

        Flight flight = new Flight("Atheist");
        flights.add(flight);
        flight = new Flight("Agnostic");
        flights.add(flight);
        flight = new Flight("Buddist");
        flights.add(flight);
        flight = new Flight("Christian");
        flights.add(flight);
        flight = new Flight("Catholic");
        flights.add(flight);
        flight = new Flight("Hindu");
        flights.add(flight);
        flight = new Flight("Islamic");
        flights.add(flight);
        flight = new Flight("Jewish");
        flights.add(flight);
        flight = new Flight("Other");
        flights.add(flight);

        return flights;
    }
}

File: app/src/main/java/com/reelme/app/view/RelegionActivity.kt
package com.reelme.app.view

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.Gravity
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import com.google.gson.Gson
import com.reelme.app.R
import com.reelme.app.adapter.RelegiousRecyclerViewAdapter
import com.reelme.app.databinding.RelegiousLayoutBinding
import com.reelme.app.listeners.EmptyResultListener
import com.reelme.app.pojos.ReligiousBelief
import com.reelme.app.pojos.UserModel
import com.reelme.app.util.GenericValues
import com.reelme.app.utils.FirbaseWriteHandlerActivity
import java.util.*

class RelegionActivity : AppCompatActivity() {
    lateinit var activity  : Activity
    private var adapter: RelegiousRecyclerViewAdapter? = null
    private var binding: RelegiousLayoutBinding? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.relegious_layout)
        binding!!.flightsRv.layoutManager = LinearLayoutManager(this)
        binding!!.flightsRv.addItemDecoration(
                DividerItemDecoration(this, DividerItemDecoration.VERTICAL))
        adapter = RelegiousRecyclerViewAdapter(prepareData(), this)
        binding!!.flightsRv.adapter = adapter
        activity = this

        binding!!.nextBtn.setOnClickListener {
            if(adapter!!.getSelectedItem()>=0) {
                userDetails.religiousBeliefs = prepareData()[adapter!!.getSelectedItem()].religious
                setUserInfo()
            }
        }

        binding!!.skipBtn.setOnClickListener {
            userDetails.skipReligiousBeliefs = true
            setUserInfo()
        }

        getUserInfo()

    }

    fun prepareData(): List<ReligiousBelief> {

        val values = GenericValues().getFileString("religion.json", this)
        return GenericValues().getRelegionList(values,this)[0].chapters!!

    }

    lateinit var userDetails : UserModel
    private var isEdit = false;

    private fun getUserInfo() {
        val sharedPreference = getSharedPreferences("AUTH_INFO", Context.MODE_PRIVATE)
        val coronaJson = sharedPreference.getString("USER_INFO", "");
        isEdit = sharedPreference.getBoolean("IS_EDIT",false)

        try {
            val auth = Gson().fromJson(coronaJson, UserModel::class.java)
            Log.d("Authentication token", auth.emailId)
            userDetails = (auth as UserModel)
        } catch (e: java.lang.Exception) {
            Log.d("Authenticaiton token", "Exception")
        }
    }

    private fun setUserInfo(){

        binding!!.progressbar.visibility= View.VISIBLE
        val gsonValue = Gson().toJson(userDetails)
        val sharedPreference =  getSharedPreferences("AUTH_INFO", Context.MODE_PRIVATE)
        val editor = sharedPreference.edit()
        editor.putString("USER_INFO",gsonValue)
        editor.putBoolean("IS_EDIT",false)
        editor.apply()

        FirbaseWriteHandlerActivity(this).updateUserInfo(userDetails, object : EmptyResultListener {
            override fun onSuccess() {
                binding!!.progressbar.visibility= View.INVISIBLE
                if(isEdit){
                    finish()
                } else{
                    startActivity(Intent(this@RelegionActivity, FragmentHobbies::class.java));
                }

                Log.d("Authenticaiton token", "onSuccess")
                Toast.makeText(activity, "we have successfully saved your profile", Toast.LENGTH_LONG).apply {setGravity(Gravity.TOP, 0, 0); show() }

            }

            override fun onFailure(e: Exception) {
                binding!!.progressbar.visibility= View.INVISIBLE
                //   fragmentSignin.startActivity(Intent(fragmentSignin, FragmentHomePage::class.java));
                Log.d("Authenticaiton token", "Exception"+e)
                Toast.makeText(activity, "Failed to save your profile.. please try again later", Toast.LENGTH_LONG).apply {setGravity(Gravity.TOP, 0, 0); show() }

            }
        })
    }


}